packages/core/strapi/lib/types/factories.d.ts

```diff
@@ -28,11 +40,12 @@ type CollectionTypeRouterConfig = {
   delete?: HandlerConfig;
 };
 
-type RouterConfig = {
+type RouterConfig<T> = {
```

Comment:
Sorry, last thing, this should now default `T = SingleTypeRouterConfig | CollectionTypeRouterConfig` ?